### PR TITLE
migrated some tests of settings from ui to ui_airgun

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
### PR objective 
Migrate Satellite settings component tests from ui to ui_airgun, currently, there is no coverage running tests from UI front.    

### Test Result 
```
(env) [root@okhatavk robottelo]# pytest tests/foreman/ui_airgun/test_settings.py -v
========================================================= test session starts ==========================================================
platform linux -- Python 3.4.8, pytest-4.0.2, py-1.7.0, pluggy-0.9.0 -- /home/okhatavk/Satellite/robottelo/env/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.26.1, services-1.3.1, mock-1.10.0, forked-1.0.2, cov-2.6.1
collecting ... 2019-06-04 11:24:11 - conftest - DEBUG - Collected 24 test cases

collected 24 items                                                                                                                     

tests/foreman/ui_airgun/test_settings.py::test_positive_update_login_page_footer_text_with_long_string SKIPPED                   [  4%]
tests/foreman/ui_airgun/test_settings.py::test_positive_remove_login_page_footer_text SKIPPED                                    [  8%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_authorize_login_delegation_param PASSED                           [ 12%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_administrator_param PASSED                                        [ 16%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_authorize_login_delegation_api_param PASSED                       [ 20%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_entries_per_page_param PASSED                                     [ 25%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_entries_per_page_param PASSED                                     [ 29%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_email_reply_address_param PASSED                                  [ 33%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_fix_db_cache_param PASSED                                         [ 37%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_max_trend_param PASSED                                            [ 41%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_max_trend_param PASSED                                            [ 45%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_idle_timeout_param PASSED                                         [ 50%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_idle_timeout_param PASSED                                         [ 54%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_foreman_url_param PASSED                                          [ 58%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_foreman_url_param PASSED                                          [ 62%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_login_page_footer_text PASSED                                     [ 66%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_dynflow_enable_console_param PASSED                               [ 70%]
tests/foreman/ui_airgun/test_settings.py::test_negative_settings_access_to_non_admin SKIPPED                                     [ 75%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_email_delivery_method_smtp SKIPPED                                [ 79%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_email_delivery_method_smtp SKIPPED                                [ 83%]
tests/foreman/ui_airgun/test_settings.py::test_positive_update_email_delivery_method_sendmail SKIPPED                            [ 87%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_email_delivery_method_sendmail SKIPPED                            [ 91%]
tests/foreman/ui_airgun/test_settings.py::test_positive_email_yaml_config_precedence SKIPPED                                     [ 95%]
tests/foreman/ui_airgun/test_settings.py::test_negative_update_hostname_with_empty_fact SKIPPED                                  [100%]


====================================== 15 passed, 9 skipped in 2395.24 seconds ======================================

```

### Note 
Dependant airgun pr is here : https://github.com/SatelliteQE/airgun/pull/334

### CODE REVIEW

File: tests/foreman/ui_airgun/test_settings.py [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+from random import choice, randint
+
+from fauxfactory import gen_email, gen_url
+from pytest import raises
```
Comment (ntkathole) [inline tests/foreman/ui_airgun/test_settings.py]: Import them in order

**Comment by ntkathole** [comment]
@omkarkhatavkar good to remove tests from UI directory

File: tests/foreman/ui_airgun/test_settings.py [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+        5. Check error msg "Unable to send email, check server log for more
+            information" is shown
+        6. Check /var/log/foreman/production.log has error msg related
+            to email
+
+    :expectedresults: Email is not sent through SMTP
+
+    :CaseImportance: Critical
+
+    :CaseLevel: Acceptance
```
Comment (ntkathole) [inline tests/foreman/ui_airgun/test_settings.py]: Please add :CaseAutomation: notautomated for tests which is not automated. May be earlier it was not there, but we should have it now.

**Comment by omkarkhatavkar** [comment]
> @omkarkhatavkar good to remove tests from UI directory

@ntkathole @oshtaier while writing these tests some of the tests are redundant as they doing the same things. I was thinking to have one End to End test and remove a redundant test, please take a look on the tests to let me know if you think the same.

**Comment by vijay8451** [comment]
> > @omkarkhatavkar good to remove tests from UI directory
> 
> @ntkathole @oshtaier while writing these tests some of the tests are redundant as they doing the same things. I was thinking to have one End to End test and remove a redundant test, please take a look on the tests to let me know if you think the same.

@omkarkhatavkar +1 write one end to end test case.

**Comment by omkarkhatavkar** [comment]
> > > @omkarkhatavkar good to remove tests from UI directory
> > 
> > 
> > @ntkathole @oshtaier while writing these tests some of the tests are redundant as they doing the same things. I was thinking to have one End to End test and remove a redundant test, please take a look on the tests to let me know if you think the same.
> 
> @omkarkhatavkar +1 to write one end to end test case.

Added the One End to End Scenario.

**Comment by omkarkhatavkar** [comment]
> @omkarkhatavkar good to remove tests from UI directory

@ntkathole : https://github.com/SatelliteQE/robottelo/pull/7012 PR will take care of it, I don't think I have to work on this separately.

**Comment by omkarkhatavkar** [comment]
@ntkathole @vijay8451 @oshtaier Can you please review this as well, whenever you get time as underneath airgun PR got merged.

File: tests/foreman/ui_airgun/test_settings.py [hunk 3]
```diff
@@ -172,7 +174,7 @@ def test_positive_update_restrict_composite_view(session, set_original_property_
 
 
 @tier1
```
Comment (vijay8451) [inline tests/foreman/ui_airgun/test_settings.py]: This is still a tier1 test , please check

File: tests/foreman/ui_airgun/test_settings.py [hunk 4]
```diff
@@ -193,154 +195,19 @@ def test_positive_update_authorize_login_delegation_param(session, set_original_
...
-                'name = {}'.format(property_name),
-                param_value
-            )
-            result = session.settings.read('name = {}'.format(property_name))
-            assert result['table'][0]['Value'] == param_value
-
-
-@tier1
-def test_negative_update_entries_per_page_param(session, set_original_property_value):
+@tier2
```
Comment (vijay8451) [inline tests/foreman/ui_airgun/test_settings.py]: It more looks a tier1 test

File: tests/foreman/ui_airgun/test_settings.py [hunk 5]
```diff
@@ -351,104 +218,15 @@ def test_negative_update_max_trend_param(session, set_original_property_value):
...
-                'name = {}'.format(property_name),
-                param_value
-            )
-            result = session.settings.read('name = {}'.format(property_name))
-            assert result['table'][0]['Value'] == param_value
-
-
-@tier1
-def test_negative_update_foreman_url_param(session, set_original_property_value):
+@tier2
```
Comment (vijay8451) [inline tests/foreman/ui_airgun/test_settings.py]: this more looks a tier1 test.

**Comment by vijay8451** [comment]
tier3 test looks good to me .

File: tests/foreman/ui_airgun/test_settings.py (hunk 4 above)
Comment (oshtaier) [reply tests/foreman/ui_airgun/test_settings.py]: I would leave only one negative scenario only. We should not cover negative scenario at all, but let's keep one

File: tests/foreman/ui_airgun/test_settings.py (hunk 3 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_settings.py]: Yes, it is a tier1 test. what do you recommend?

Comment (vijay8451) [reply tests/foreman/ui_airgun/test_settings.py]: under UI , we do not create tier1 tests and covering more under one `API/CLI` test only.
Do you think same type of functionality covered under exists `tier2/3` tests ? if yes then good to remove this one and if no then would say to cover under any `tier2/3` for same.

Comment (oshtaier) [reply tests/foreman/ui_airgun/test_settings.py]: as you can see, we have zero tier1 in web ui, because it is a requirement to new framework.

Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_settings.py]: OK I will remove this test, as per your comments mentioned above.

File: tests/foreman/ui_airgun/test_settings.py (hunk 4 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_settings.py]: @oshtaier @vijay8451 As this a negative test and tier1 test. As per your earlier comments, I will be removing this test as well. Please let me know if you have questions

File: tests/foreman/ui_airgun/test_settings.py (hunk 5 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_settings.py]: @oshtaier @vijay8451 As this a negative test and tier1 test. As per your earlier comments, I will be removing this test as well. Please let me know if you have questions

File: tests/foreman/ui_airgun/test_settings.py (hunk 4 above)
Comment (vijay8451) [reply tests/foreman/ui_airgun/test_settings.py]: @omkarkhatavkar I would say to remove all tier1 tests here ... but as mentioned by @oshtaier he is good to keep one negative test .

@oshtaier what do you say on it ? as we could cover negative tests under `API/CLI` around tier1 ?

File: tests/foreman/ui_airgun/test_settings.py (hunk 5 above)
Comment (vijay8451) [reply tests/foreman/ui_airgun/test_settings.py]: same as above

File: tests/foreman/ui_airgun/test_settings.py (hunk 4 above)
Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_settings.py]: @vijay8451 Just to add here this, such tests (invalid input/validation tests) should be handled in a unit test level. If you consider this test this test validate the UI Error for invalid input (So intention to see javascript error after an ajax call exception ). Certainly, this is cannot be validated in API/CLI.

Comment (vijay8451) [reply tests/foreman/ui_airgun/test_settings.py]: @omkarkhatavkar right , then as per my understanding we should keep one nagative test and keep it with  `tier2`  marker (as exception one by mentioning in doc string) ... what do you think ?
@oshtaier ^^

Comment (omkarkhatavkar) [reply tests/foreman/ui_airgun/test_settings.py]: @vijay8451 @oshtaier we do have tier2 tests which are actually tier1 test similar to here https://github.com/omkarkhatavkar/robottelo/blob/9e5c7f97831c58e3ec8a27639a13c89b54bc37ad/tests/foreman/ui_airgun/test_contentview.py#L865

Comment (oshtaier) [reply tests/foreman/ui_airgun/test_settings.py]: that is right, we have few rare situation, so yeah, keep tier2 marker for that one case

Comment (vijay8451) [reply tests/foreman/ui_airgun/test_settings.py]: @omkarkhatavkar then I believe not an issue to add one negative test :+1:

**Comment by oshtaier** [comment]
don't see any runs since beginning of PR, especially for `test_positive_update_restrict_composite_view`

ACK pending one test run. Anyone can merge PR after it is provided

**Comment by omkarkhatavkar** [comment]
@oshtaier 
```
(env) [root@okhatavk robottelo]# pytest tests/foreman/ui_airgun/test_settings.py 
=============================================== test session starts ===============================================
platform linux -- Python 3.4.8, pytest-4.0.2, py-1.7.0, pluggy-0.9.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.26.1, services-1.3.1, mock-1.10.0, forked-1.0.2, cov-2.6.1
collecting ... 2019-06-18 14:06:55 - conftest - DEBUG - Collected 10 test cases

collected 10 items                                                                                                

tests/foreman/ui_airgun/test_settings.py ..ssssssss                                                                [100%]

================================================ warnings summary =================================================
tests/foreman/ui_airgun/test_settings.py::test_positive_update_restrict_composite_view
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:130: DeprecationWarning: The value of convert_charrefs will become True in 3.5. You are encouraged to set the value explicitly.
    parser = html_parser.HTMLParser()
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:133: DeprecationWarning: The unescape method is deprecated and will be removed in 3.5, use html.unescape() instead.
    for option
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:130: DeprecationWarning: The value of convert_charrefs will become True in 3.5. You are encouraged to set the value explicitly.
    parser = html_parser.HTMLParser()
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/widgetastic/widget/select.py:133: DeprecationWarning: The unescape method is deprecated and will be removed in 3.5, use html.unescape() instead.
    for option

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================ 2 passed, 8 skipped, 4 warnings in 335.03 seconds ================================

```

**Pull request merged**